File: app/script/speak.js
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   speak.js                                           :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: adjivas <adjivas@student.42.fr>            +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2014/09/15 10:02:36 by adjivas           #+#    #+#             */
/*   Updated: 2014/09/15 10:02:36 by adjivas          ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

'use strict';

var py = require('python-shell');

/* 
** The Speak's class is calls for request the socket of
** says a word.
*/

var Speak = {
  'silent': !Configuration.synthse.active,

  'run': function (word) {
    if (!Speak.silent) {
      py.run('speak.py', {
        'scriptPath': './app/python/',
        'pythonPath': Configuration.python.path,
        'args': [word]
      }, function (err, results) {
        if (err)
          Debug.console(err);
      });
    }
  },
  'call': function (arg) {
    var elemt = document.querySelector(Menu.find);

    elemt = elemt.tagName.toLowerCase();
    elemt = Lang.translate(elemt, {});
    Speak.run(elemt);
  }
};

File: app/python/speak.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    speak.py                                           :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: adjivas <adjivas@student.42.fr>            +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2014/09/15 10:02:36 by adjivas           #+#    #+#              #
#    Updated: 2014/09/15 10:02:36 by adjivas          ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

import speech
import sys

class Speak:
  def __init__(self, ac, av):
    ct = 1;

    while (ct <= ac):
     speech.say(av[ct]);
     ct += 1;

if (__name__ == '__main__'):
  Speak(len(sys.argv) - 1, sys.argv);
